examples/editor-assignment/src/locations/ConfigScreen.tsx

```diff
@@ -0,0 +1,200 @@
+    height: 'auto',
+    minHeight: '65vh',
+    margin: '0 auto',
+    marginTop: tokens.spacingXl,
+    padding: `${tokens.spacingXl} ${tokens.spacing2Xl}`,
+    maxWidth: tokens.contentWidthText,
+    backgroundColor: tokens.colorWhite,
+    zIndex: 2,
+    boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.1)',
+    borderRadius: '2px',
+  }),
+  background: css({
+    display: 'block',
+    position: 'absolute',
+    zIndex: -1,
+    top: 0,
+    width: '100%',
+    height: '300px',
+    backgroundColor: tokens.colorPrimary,
+  }),
+};
+
+const buildSidebarTargetState = (selectedSidebarCTs: string[]) => {
+  return selectedSidebarCTs.reduce(
+    (acc, ct) => ({
+      ...acc,
+      [ct]: {
+        sidebar: { position: 0 },
+      },
+    }),
+    {}
+  );
+};
+
+const onCTSelect =
+  (selectedCTs: string[], setSelectedCTs: (cts: string[]) => void) => (ct: string) => {
+    selectedCTs.includes(ct)
+      ? setSelectedCTs(selectedCTs.filter((item) => item !== ct))
+      : setSelectedCTs([...selectedCTs, ct]);
+  };
+
+const ConfigScreen = () => {
+  const [parameters, setParameters] = useState<AppInstallationParameters>({});
+  const [contentTypes, setContentTypes] = useState<ContentTypeProps[]>([]);
+  const [selectedSidebarCTs, setSelectedSidebarCTs] = useState<string[]>([]);
+  const [supportedFieldCTs, setSupportedFieldCTs] = useState<ContentTypeProps[]>([]);
+  const [selectedFieldCTs, setSelectedFieldCTs] = useState<string[]>([]);
+  const sdk = useSDK<AppExtensionSDK>();
+  const cma = useCMA();
+
+  const onConfigure = useCallback(async () => {
+    // This method will be called when a user clicks on "Install"
+    // or "Save" in the configuration screen.
+    // for more details see https://www.contentful.com/developers/docs/extensibility/ui-extensions/sdk-reference/#register-an-app-configuration-hook
+
+    // Get current the state of EditorInterface and other entities
+    // related to this app installation
+    const currentState = await sdk.app.getCurrentState();
+
+    return {
+      // Parameters to be persisted as the app configuration.
+      parameters,
+      // In case you don't want to submit any update to app
+      // locations, you can just pass the currentState as is
+      targetState: {
+        EditorInterface: {
+          ...currentState?.EditorInterface,
+
+          // apply to all selected content types' sidebar & fields
+          ...merge(
+            buildSidebarTargetState(selectedSidebarCTs),
+            buildFieldTargetState(selectedFieldCTs, supportedFieldCTs)
+          ),
+        },
+      },
+    };
+  }, [parameters, sdk, selectedSidebarCTs, selectedFieldCTs, supportedFieldCTs]);
+
+  const onSidebarContentTypeClick = useCallback(
+    onCTSelect(selectedSidebarCTs, setSelectedSidebarCTs),
+    [selectedSidebarCTs, setSelectedSidebarCTs]
+  );
+  const onFieldContentTypeClick = useCallback(onCTSelect(selectedFieldCTs, setSelectedFieldCTs), [
+    selectedFieldCTs,
+    setSelectedFieldCTs,
+  ]);
```

Comment:
That's not how you should use `useCallback` and this causes the lint errors in the review comment